crates/generated_parser/src/ast_builder.rs

```diff
@@ -1436,36 +1436,40 @@ impl<'alloc> AstBuilder<'alloc> {
         super_token: arena::Box<'alloc, Token>,
         expression: arena::Box<'alloc, Expression<'alloc>>,
         close_token: arena::Box<'alloc, Token>,
-    ) -> arena::Box<'alloc, Expression<'alloc>> {
+    ) -> Result<'alloc, arena::Box<'alloc, Expression<'alloc>>> {
+        self.check_super()?;
+
```

Comment:
Could you hold this patch out-side of this PR, and post-pone until @codehag finishes with the split between the early errors and the AST Builder?